src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -192,14 +193,30 @@ class BookingExtractionService(
     if (mostRecentSentencesByReleaseDate.any { it.isRecall() }) {
       dates[PRRD] = latestReleaseDate
     }
-    if (booking.sentences.all { it is DetentionAndTrainingOrderSentence }) {
+    if (booking.sentences.any { it is DetentionAndTrainingOrderSentence }) {
       dates[MTD] = latestReleaseDate
-      if (mostRecentSentenceByExpiryDate.releaseDateTypes.contains(ETD)) {
-        calculateEtd(mostRecentSentenceByExpiryDate, dates)
-      }
-
-      if (mostRecentSentenceByExpiryDate.releaseDateTypes.contains(LTD)) {
-        calculateLtd(mostRecentSentenceByExpiryDate, dates)
+      calculateWhenAllDtos(mostRecentSentenceByExpiryDate, dates)
+      if (sentences.size > 1 && !sentences.any { it is ConsecutiveSentence } && !sentences.all { it is DetentionAndTrainingOrderSentence }) {
+        val firstNonDtoSentence = sentences.first { it !is DetentionAndTrainingOrderSentence }
+        val firstDtoSentence = sentences.first { it is DetentionAndTrainingOrderSentence }
+        val type = if (firstNonDtoSentence.releaseDateTypes.contains(CRD)) CRD else ARD
```

Comment:
I think this should probably be `concurrentOraAndNonOraDetails.isReleaseDateConditional`